plugins/apache-airflow/src/api/ApacheAirflowClient.ts

```diff
@@ -75,6 +75,25 @@ export class ApacheAirflowClient implements ApacheAirflowApi {
     return dags;
   }
 
+  async getDags(
+    dagIds: string[],
+  ): Promise<{ dags: Dag[]; dagsNotFound: string[] }> {
+    const dagsNotFound: string[] = [];
+    const response = await Promise.all(
+      dagIds.map(id => {
+        return this.fetch<Dag>(`/dags/${id}`).catch(e => {
+          if (e.message.toUpperCase('en-US') === 'NOT FOUND') {
```

Comment:
Looks like your tests set status 404 which at least when I test locally does not throw and should instead be available as `.status` on the response object. Does jenkins return a 404 or 500(which throws)?

Reply:
From the `ApacheAirflowClient` private `fetch` wrapper
```typescript
const response = await fetch(`${proxyUri}${input}`, init);
if (!response.ok) throw new Error(response.statusText);
```

So the 404 is automatically converted into a `Not Found` error. I don't have access to the `.status`